js/google_ima.js

```diff
@@ -524,13 +526,33 @@ require("../html5-common/js/utils/utils.js");
       });
 
       /**
-       * Called by the ad manager controller.  Ad Manager Controller lets the module know that an ad should play now.
-       * @public
+       * debouncing the case when user forwarded ahead and 2 or more ads requests simultaneosly issued
+       * @see https://jira.corp.ooyala.com/browse/PLAYER-3832
+       * @private
        * @method GoogleIMA#playAd
-       * @param {object} ad The ad to play from the timeline.
+       * @param {object} ad The ad to play from the timeline
+       * @returns {void}
        */
+      let debouncerId = null;
       this.playAd = function(amcAdPod)
       {
+        if (debouncerId) {
+          _resetAdsState();
+        }
+        debouncerId = setTimeout(() => {
+          debouncerId = null;
+          playAd(amcAdPod);
+        }, DEBOUNCE_TIMEOUT);
```

Comment:
Pasting my comment from the JIRA ticket:

Is this adding 500 ms to the user experience prior to an ad playback (even a single ad playback), or will this only be used if there are multiple requests in quick succession? If for a single preroll/ad there is now an additional 500 ms of latency, we cannot use this fix/workaround. That adds a significant time to time to first frame for prerolls and when transitioning from content to ads.

Reply:
@zelibobla please take a look at this comment. This is a major issue if we are adding 500 ms prior to every ad playback.

Reply:
Yep, we had to have a call for this, but due to another priority this wasn't discussed. Just forgot then.
At the moment when user clicked forward and got over the ads (and we insist him to see the ads), we yet don't know if he will click immediately again and get over next ads. This is why we need debounce timeout. And yes: we have to use it even for single ads.
It hits UX yes, as I explained originally before implementing this.